docs/guides/circuits.rst

```diff
@@ -0,0 +1,439 @@
+  ic| len(qaoa_circ.operations): 15
+  ic| op.gate.name: 'H', op.qubit_indices: (0,), op.params: ()
+  ic| op.gate.name: 'H', op.qubit_indices: (1,), op.params: ()
+  ic| op.gate.name: 'H', op.qubit_indices: (2,), op.params: ()
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (0, 1), op.params: ()
+  ic| op.gate.name: 'RZ', op.qubit_indices: (1,), op.params: (beta_0,)
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (0, 1), op.params: ()
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (0, 2), op.params: ()
+  ic| op.gate.name: 'RZ', op.qubit_indices: (2,), op.params: (beta_1,)
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (0, 2), op.params: ()
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (1, 2), op.params: ()
+  ic| op.gate.name: 'RZ', op.qubit_indices: (2,), op.params: (beta_2,)
+  ic| op.gate.name: 'CNOT', op.qubit_indices: (1, 2), op.params: ()
+  ic| op.gate.name: 'RX', op.qubit_indices: (0,), op.params: (gamma_0,)
+  ic| op.gate.name: 'RX', op.qubit_indices: (1,), op.params: (gamma_1,)
+  ic| op.gate.name: 'RX', op.qubit_indices: (2,), op.params: (gamma_2,)
+
+Powers of gates and gate exponentials
+-------------------------------------
+
+You can add a power of arbitrary non-symbolic gates to your circuit, as well as get the gate exponential. This won't be used in the creation of our QAOA circuit, but we'll still provide examples here. 
+
+**To raise a gate to a power**, you can use ``.power(exponent)`` with the ``exponent`` argument the power you want to raise the gate to. Here we verify that the Hadamard gate raised to the 2nd power is the same as the identity.
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: ic(H.power(2).matrix == sympy.eye(2))
+  :end-at: ic(H.power(2).matrix == sympy.eye(2))
+
+Where the output is ``ic| H.power(2).matrix == sympy.eye(2): True``
+
+**To get the gate exponential** of a specific gate, you can use ``.exp``, shown in this example with ``.matrix`` to view what the matrix looks like. It is also generally a good idea to simplify the matrix when possible.
+
+.. literalinclude:: /examples/circuits_guide.py
+  :language: python
+  :start-at: ic(sympy.simplify(H.exp.matrix))
+  :end-at: ic(sympy.simplify(H.exp.matrix))
+
+
+.. _wavefunction_operations:
+
+Gate operations vs Wave Function Operations
+-------------------------------------------
+
+While gate operations perform the familiar gates on the qubits of a circuit, `wavefunction operations <https://github.com/zapatacomputing/orquestra-quantum/blob/main/src/orquestra/quantum/circuits/_wavefunction_operations.py>`_ can operate on the wavefunction directly. This is convenient if the operation you want to perform is not easily expressible using gates or is not unitary at all. Currently, the only built-in wavefunction operation in Orquestra is the ``MultiPhaseOperation``, which allows a specific phase (given as an angle theta) to be applied to all 2^N components of the wavefunction for an N-qubit circuit.
```

Comment:
I'd add info that you can use `WavefunctionOperation` to quickly prepare your simulation in a given state if you know the amplitude vector of this state.

Reply:
fixed in af446a1749110a289630084291990551053306a0